src/Psl/Internal/suppress.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Internal;
+
+use function error_reporting;
+
+/**
+ * @psalm-template T
+ *
+ * @psalm-param (callable(): T) $fun
+ *
+ * @psalm-return T
+ */
+function suppress(callable $fun)
+{
+    $level = error_reporting(0);
+
+    $result = $fun();
```

Comment:
This should probably be wrapped in a `try` `finally`. Otherwise we may end up having `error_reporting` on 0 for the rest of the process.

e.g.

```php
$level = error_reporting(0);

try {
    $result = $fun();
} finally {
    error_reporting($level);
}

return $result;
```